app/Models/Backup.php

```
@@ -0,0 +1,47 @@
+<?php
+
+namespace App\Models;
+
+
+
+class Backup
+{
+    public function store($destinationPath, $directory)
+    {
+        if (!is_dir($directory)) {
+            return redirect()
+                ->back()
+                ->withErrors(["backupPath" => "Le chemin de sauvegarde spécifié n'existe pas ou n'est pas un répertoire valide."]);
+        }
+
+        try {
+            copy(base_path("database/database.sqlite"), $destinationPath);
+        } catch (\Exception $e) {
```

avoid catch Exception it will catch all exception try be more precise when catching exception.

---

by reading doc 
https://www.php.net/manual/en/function.copy.php 
i don't see any exception throwed by copy